serialization/parse_node_helper.go

```diff
@@ -0,0 +1,22 @@
+package serialization
+
+import "errors"
+
+// MergeDeserializersForIntersectionWrapper merges the given fields deserializers for an intersection type into a single collection.
+func MergeDeserializersForIntersectionWrapper(targets ...Parsable) (map[string]func(ParseNode) error, error) {
+	if len(targets) == 0 {
+		return nil, errors.New("no targets provided")
+	}
+	if len(targets) == 1 {
+		return targets[0].GetFieldDeserializers(), nil
+	}
+	deserializers := make(map[string]func(ParseNode) error)
+	for _, target := range targets {
+		for key, deserializer := range target.GetFieldDeserializers() {
+			if _, ok := deserializers[key]; !ok {
+				deserializers[key] = deserializer
+			}
+		}
+	}
+	return deserializers, nil
```

Comment:
How do we intend to handle collisions? A `map[string][]func()` perhaps in this case

Common properties on different composed types might result in type erasure

```yml
components:
  schemas:
    Dog:
      type: object
      properties:
        bark:
          type: boolean
        breed:
          type: string
          enum: [Dingo, Husky, Retriever, Shepherd]
    Cat:
      type: object
      properties:
        hunts:
          type: boolean
        age:
          type: integer
        breed:
          type: string
          enum: [Siberian, Persian]

```

Reply:
An early decision was this scenario will be a limitation of the implementation. First member type property will be the one that'll try to parse, and if the type doesn't match, the information will go in the additional data. If we receive significant feedback about that, we might first instruct people to avoid collisions in their descriptions, if the feedback becomes "this is not good enough" we'll implement a different way to merge that so both field deserializers are called in the future.